fairlearn/metrics/_metric_frame.py

```diff
@@ -3,32 +3,40 @@
 
 import copy
 import logging
+import warnings
+from functools import wraps
+from typing import Any, Callable, Dict, List, Optional, Union
+
 import numpy as np
 import pandas as pd
-from typing import Any, Callable, Dict, List, Optional, Union
 from sklearn.utils import check_consistent_length
-import warnings
-from functools import wraps
 
 from fairlearn.metrics._input_manipulations import _convert_to_ndarray_and_squeeze
 from ._function_container import FunctionContainer, _SAMPLE_PARAMS_NOT_DICT
 from ._group_feature import GroupFeature
 
-
 logger = logging.getLogger(__name__)
 
 _SUBGROUP_COUNT_WARNING_THRESHOLD = 20
 
-_SF_DICT_CONVERSION_FAILURE = "DataFrame.from_dict() failed on sensitive features. " \
-    "Please ensure each array is strictly 1-D."
+_SF_DICT_CONVERSION_FAILURE = ("DataFrame.from_dict() failed on sensitive features. "
+                               "Please ensure each array is strictly 1-D.")
 _BAD_FEATURE_LENGTH = "Received a feature of length {0} when length {1} was expected"
 _SUBGROUP_COUNT_WARNING = "Found {0} subgroups. Evaluation may be slow"
 _FEATURE_LIST_NONSCALAR = "Feature lists must be of scalar types"
 _FEATURE_DF_COLUMN_BAD_NAME = "DataFrame column names must be strings. Name '{0}' is of type {1}"
 _DUPLICATE_FEATURE_NAME = "Detected duplicate feature name: '{0}'"
 _TOO_MANY_FEATURE_DIMS = "Feature array has too many dimensions"
-_SAMPLE_PARAM_KEYS_NOT_IN_FUNC_DICT = \
-    "Keys in 'sample_params' do not match those in 'metric'"
+_SAMPLE_PARAM_KEYS_NOT_IN_FUNC_DICT = ("Keys in 'sample_params'"
+                                       " do not match those in 'metric'")
+_BAD_INIT_ERROR = """Non streaming metrics require data to process.
+ Set streaming=True for streaming metrics"""
```

Comment:
we should be consistent, either have newlines in all messages or none, I prefer none.